src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -366,29 +345,18 @@ protected String processACHBatchDetail(PayeeACHAccountExtractDetail achDetail) {
             return processingError;
         }
         
-        processingError = addOrUpdateACHAccountIfNecessary(payee, achDetail, PayeeIdTypeCodes.ENTITY, payee.getEntityId());
+        processingError = payeeACHAccountDocumentService.addOrUpdateACHAccountIfNecessary(
+                payee, achDetail, PayeeIdTypeCodes.ENTITY, payee.getEntityId());
         if (StringUtils.isNotBlank(processingError)) {
             return processingError;
         }
         
-        processingError = addOrUpdateACHAccountIfNecessary(payee, achDetail, PayeeIdTypeCodes.EMPLOYEE, payee.getEmployeeId());
+        processingError = payeeACHAccountDocumentService.addOrUpdateACHAccountIfNecessary(
+                payee, achDetail, PayeeIdTypeCodes.EMPLOYEE, payee.getEmployeeId());
         
         return processingError;
     }
 
-    protected String addOrUpdateACHAccountIfNecessary(
-            Person payee, PayeeACHAccountExtractDetail achDetail, String payeeType, String payeeIdNumber) {
-        GlobalVariables.getMessageMap().clearErrorMessages();

```

Comment:
would it be worth clearing error messages in the new version of this function, in case any down steam processing expected this function to clear any previous errors?
I know the new version makes the achService call in a new global variables, but maybe it would be good to clear errors for other processing?

Reply:
The old code was making this `clearErrorMessages()` call prior to each attempted ACH account insertion or update, so I don't see what other downstream processing would need to have the error messages retained beyond the insert/update attempt. Also, there are only a small number of areas in base code that are calling `clearErrorMessages()` directly (likely because they haven't been refactored to use the do-in-new-variables approach), so I would advise against including this direct call unless you know of a specific use case.